iocore/hostdb/HostDB.cc

```diff
@@ -449,13 +449,13 @@ HostDBContinuation::init(HostDBHash const &the_hash, Options const &opt)
 void
 HostDBContinuation::refresh_hash()
 {
-  ProxyMutex *old_bucket_mutex = hostDB.refcountcache->lock_for_key(hash.hash.fold());
+  Ptr<ProxyMutex> old_bucket_mutex = hostDB.refcountcache->lock_for_key(hash.hash.fold());
   // We're not pending DNS anymore.
   remove_trigger_pending_dns();
   hash.refresh();
   // Update the mutex if it's from the bucket.
   // Some call sites modify this after calling @c init so need to check.
-  if (mutex.get() == old_bucket_mutex) {
+  if (mutex.get() == old_bucket_mutex.get()) {
```

Comment:
You should be able to drop the `get` here and do `mutex == old_bucket.mutex`. The `Ptr` class overloads the equality operator to compare the internal pointers.